### ISSUE TITLE
onBindListener overload for lambdas

### ISSUE THREAD

**Issue by sargunster** [nitrico/LastAdapter#2]
Having to do this is ugly:
```kotlin
        LastAdapter.with(content, BR.item)
                .map<ListItem>(R.layout.fragment_grid_list_item)
                .onBindListener(object : LastAdapter.OnBindListener {
                    override fun onBind(item: Any, view: View, pos: Int) {
                        doStuff()
                    }
                })
                .into(contentView)
```

I would much rather do this:
```kotlin
        LastAdapter.with(content, BR.item)
                .map<ListItem>(R.layout.fragment_grid_list_item)
                .onBindListener { item, view, pos -> doStuff() }
                .into(contentView)
```

Also, similar for click and long click listeners.

I can provide the PR if you approve this feature request.

**Comment by nitrico** [comment]
I totally agree and that was my intention but I realized that if the library was written in Java that would be automatic but it isn't when written in Kotlin... 

Please, send the PR.

**Comment by nitrico** [comment]
I'm about to update the library with this https://gist.github.com/nitrico/90e15f235782e1d8b82b2567116a3a5f

The usage then would be like that:

```kotlin
LastAdapter.with(Data.items, BR.item)
           .layout { item, pos -> getItemLayout(item, pos) }
           .onBind { item, view, pos -> println("onBind position $pos: $item") }
           .onClick { item, view, pos -> println("onClick position $pos: $item") }
           .onLongClick { item, view, pos -> println("onLongClick position $pos: $item") }
           .into(recyclerView)
```

I did it for the LayoutHandler as well.

Let me know if your solution was cleaner before I upload it.

**Comment by sargunster** [comment]
Mine is essentially identical :)

**Comment by nitrico** [comment]
Cool. I'm going to test it a little bit more and will update the lib soon. Thanks :)

**Comment by nitrico** [comment]
It became a bit better eventually. `item`, `view` and `position` are now available inside the lambda without declaring them on the left side.

```
LastAdapter.with(Data.items, BR.item)
           .layout { getItemLayout(item, position) }
           .onBind { println("onBind $position: $item") }
           .onClick { println("onClick $position: $item") }
           .onLongClick { println("onLongClick $position: $item") }
           .into(recyclerView)
```

Already available on version 1.1.0 through jcenter.

**Comment by sargunster** [comment]
Thanks!
